ndbench-geode-plugins/src/main/java/com/netflix/ndbench/geode/plugin/EnvParser.java

```diff
@@ -0,0 +1,91 @@
+package com.netflix.ndbench.geode.plugin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Reused from
+ * https://github.com/gemfire/cf-gemfire-connector-examples
+ *
+ * @author Pulkit Chandra
+ */
+public class EnvParser {
+  private final static Logger logger = LoggerFactory.getLogger(EnvParser.class);
+
+  private static EnvParser instance;
+  private final Pattern p = Pattern.compile("(.*)\\[(\\d*)\\]");
+
+  private EnvParser() {
+  }
+
+  public static EnvParser getInstance() {
+    if (instance != null) {
+      return instance;
+    }
+    synchronized (EnvParser.class) {
+      if (instance == null) {
+        instance = new EnvParser();
+      }
+    }
+    return instance;
+  }
+
+  public List<URI> getLocators() throws IOException, URISyntaxException {
+    List<URI> locatorList = new ArrayList<URI>();
+    Map credentials = getCredentials();
+    List<String> locators = (List<String>) credentials.get("locators");
+    for (String locator : locators) {
+      Matcher m = p.matcher(locator);
+      if (!m.matches()) {
+        throw new IllegalStateException("Unexpected locator format. expected host[port], got"+locator);
+      }
+      locatorList.add(new URI("locator://" + m.group(1) + ":" + m.group(2)));
+    }
+    return locatorList;
+  }
+
+  public String getUsername() throws IOException {
+    Map credentials = getCredentials();
+    List<Map<String, String>> userData = (List<Map<String, String>>) credentials.get("users");
+    return userData.get(0).get("username");
+  }
+
+
+  public String getPasssword() throws IOException {
+    Map credentials = getCredentials();
+    List<Map<String, String>> userData = (List<Map<String, String>>) credentials.get("users");
+    return userData.get(0).get("password");
+  }
+
+  private Map getCredentials() throws IOException {
+    Map credentials = null;
+    String envContent = System.getenv().get("VCAP_SERVICES");
```

Comment:
Environment variables are read using https://github.com/Netflix/archaius. Sample code on how to use in NdBench is @ https://github.com/Netflix/ndbench/blob/master/ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java#L62. inject it using https://github.com/Netflix/ndbench/blob/master/ndbench-cass-plugins/src/main/java/com/netflix/ndbench/plugin/cass/CassJavaDriverBatch.java#L34